examples/main/main.cpp

```diff
@@ -848,8 +872,12 @@ int main(int argc, char ** argv) {
                         string_process_escapes(buffer);
                     }
 
+                    std::string user_inp = params.conversation
+                        ? chat_add_and_format("user", buffer)
+                        : buffer;
+                    // TODO: one inconvenient of current chat template implementation is that we can't distinguish between user input and special tokens (prefix/postfix)
```

Comment:
When `params.conversation == false` there is an extra string copy that should be avoided here

Regarding the comment - can you illustrate with an example as I'm not sure what is the issue

Reply:
An example would be a prompt like this: `Which one is correct HTML tag? <s> or <a>?`

Some models having `<s>` as BOS will see the prompt as `Which one is correct HTML tag? BOS or <a>?`

Leaving `special == false` will fix that, but will also break chat template since we're now adding special tokens to user's text. This could be avoided with some more code. But IMO it's not really a big deal though, assuming that special tokens are unlikely to accidentally appear in the text.

Reply:
I added a `std::move(buffer)` since we no longer use `buffer` after this line. Is it OK to do so?

Reply:
Aha got it. Yes, for now let's make have the simple solution